src/taskpane/components/ReferenceList.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  ITheme,
+  getTheme,
+  List,
+  mergeStyleSets,
+  getFocusStyle,
+  FocusZone,
+  FocusZoneDirection,
+  Checkbox,
+} from "@fluentui/react";
+import React from "react";
+
+const theme: ITheme = getTheme();
+const { palette, semanticColors, fonts } = theme;
+
+interface ReferenceListProps {
+  list: Array<{}>;
+  onCheckBoxChange: any;
+}
+const classNames = mergeStyleSets({
+  container: {
+    overflow: "auto",
+    height: "80vh",
+    padding: "0.25rem 0.25rem 0px",
+    boxSizing: "border-box",
+  },
+  itemCell: [
+    getFocusStyle(theme, { inset: -1 }),
+    {
+      backgroundColor: theme.palette.neutralLighterAlt,
+      minHeight: 54,
+      padding: "0.25rem",
+      margin: 2,
+      boxSizing: "border-box",
+      borderBottom: `1px solid ${semanticColors.bodyDivider}`,
+      display: "flex",
+      selectors: {
+        "&:hover": { background: palette.themeLighterAlt },
+      },
+    },
+  ],
+  itemContent: {
+    marginLeft: 10,
+    boxSizing: "border-box",
+    overflow: "auto",
+    flexGrow: 1,
+  },
+  itemTitle: [
+    fonts.mediumPlus,
+    {
+      whiteSpace: "nowrap",
+      position: "relative",
+      maxHeight: "5.4em",
+      lineHeight: "1.8em",
+      overflow: "hidden",
+      textOverflow: "ellipsis",
+    },
+  ],
+  itemAuthor: {
+    fontSize: fonts.small.fontSize,
+    color: palette.neutralTertiary,
+    marginBottom: 10,
+  },
+  itemYear: {
+    fontSize: fonts.smallPlus,
+    color: palette.neutralTertiary,
+  },
+  itemType: {
+    fontSize: fonts.smallPlus,
+    color: palette.neutralTertiary,
+  },
+  checkbox: {
+    marginTop: 6,
+    marginLeft: 4,
+  },
+});
+
+function ReferenceList(props: ReferenceListProps) {
+  const onRenderCell = (item): JSX.Element => {
+    return (
+      <div className={classNames.itemCell} data-is-focusable={true}>
+        <Checkbox
+          className={classNames.checkbox}
+          title={item.title}
+          checked={!!item.isChecked}
+          onChange={props.onCheckBoxChange}
```

Comment:
I think it's an implementation detail that a checkbox is used to select items. Thus, I would prefer to have a `selectedItems` list that is automatically updated and exposed.

Reply:
@tobiasdiez  This is something I am really confused with. I looked onto some other implementation, and all of em has an extra field for the checkbox`{..., isChecked: boolean}`. Also, we only need to pass a list of reference `id` to the `citeproc` for citation and bibliography. So I added an extra field to every list entry to check whether it is selected on not and if it is checked, then update the isChecked field to true and filter over the list to get the checked entries id. But i am still not sure if it is the correct way to handle checkbox input.

Reply:
I guess there are two ways to keep track of the selected items:
- Have an  `isSelected` boolean for each item, which is toggled based on the checkbox.
- Store all selected items to a new list `selectedItems` (or maybe only store the selected indicies).

Functionality wise they are equivalent (i.e. there is nothing that you can implement in one approach but not the other). The only difference lies in the performance, and then it depends on the use case scenario which approach is better. The first one makes it very easy (and quick) to decide if a given item is selected, but costs more memory (the additional memory needed scales with the number of items). The second approach is slow and memory intensive if you have a huge list of selected items. Anyway, in our case these differences should be negligible. So it's only a matter of which approach is easier to implement.

My main point was that the reference list should manage its state completely, i.e. it should be the only place that adds the `isChecked` toggle to each item or manages the selectedItems list. Moreover, it should provide a convenient way get the selected items (either by filtering the internal list based on `isChecked` or by simply returning `selectedItems`).

Reply:
This is an example of a list with checkboxes backed by a `selectItems` list: https://github.com/hawkrives/react-list-select/blob/master/src/list.js